// passkit-generator/lib/esm/schemas/PassFieldContent.js
import Joi from "joi";
import { Semantics } from "./Semantics.js";
export const PassFieldContent = Joi.object().keys({
    attributedValue: Joi.alternatives(Joi.string().allow(""), Joi.number(), Joi.date().iso()),
    changeMessage: Joi.string(),
    dataDetectorTypes: Joi.array().items(Joi.string().regex(/(PKDataDetectorTypePhoneNumber|PKDataDetectorTypeLink|PKDataDetectorTypeAddress|PKDataDetectorTypeCalendarEvent)/, "dataDetectorType")),
    label: Joi.string().allow(""),
    textAlignment: Joi.string().regex(/(PKTextAlignmentLeft|PKTextAlignmentCenter|PKTextAlignmentRight|PKTextAlignmentNatural)/, "graphic-alignment"),
    key: Joi.string().required(),
    value: Joi.alternatives(Joi.string().allow(""), Joi.number(), Joi.date().iso()).required(),
    semantics: Semantics,
    // date fields formatters, all optionals
    dateStyle: Joi.string().regex(/(PKDateStyleNone|PKDateStyleShort|PKDateStyleMedium|PKDateStyleLong|PKDateStyleFull)/, "date style"),
    ignoresTimeZone: Joi.boolean(),
    isRelative: Joi.boolean(),
    timeStyle: Joi.string().regex(/(PKDateStyleNone|PKDateStyleShort|PKDateStyleMedium|PKDateStyleLong|PKDateStyleFull)/, "date style"),
    // number fields formatters, all optionals
    currencyCode: Joi.string().when("value", {
        is: Joi.number(),
        otherwise: Joi.string().forbidden(),
    }),
    numberStyle: Joi.string()
        .regex(/(PKNumberStyleDecimal|PKNumberStylePercent|PKNumberStyleScientific|PKNumberStyleSpellOut)/)
        .when("value", {
        is: Joi.number(),
        otherwise: Joi.string().forbidden(),
    }),
});
/**
 * @deprecated Use `PassFieldContent` instead,
 * which is the right Apple name.
 */
export const Field = PassFieldContent;
export const PassFieldContentWithRow = PassFieldContent.concat(Joi.object().keys({
    row: Joi.number().min(0).max(1),
}));
/**
 * @deprecated Use `PassFieldContentWithRow` instead,
 * which is the right Apple name.
 */
export const FieldWithRow = PassFieldContentWithRow;
