// passkit-generator/lib/esm/schemas/SemanticTagType.js
import Joi from "joi";
import { RGB_HEX_COLOR_REGEX } from "./regexps.js";
export const CurrencyAmount = Joi.object().keys({
    currencyCode: Joi.string(),
    amount: Joi.string(),
});
export const EventDateInfo = Joi.object().keys({
    date: Joi.string().isoDate(),
    ignoreTimeComponents: Joi.boolean(),
    timeZone: Joi.string(),
    /**
     * @iOSVersion 18.1
     *
     * Indicates that the time was not announced yet.
     * Leads to showing "TBA" in the UI when `date` is set.
     * Setting `ignoreTimeComponents` to true, has higher priority
     * over this property.
     *
     * When both `date` and `semantics.eventStartDate` are unset,
     * `Date: TBA` will be shown in the UI.
     */
    unannounced: Joi.boolean(),
    /**
     * @iOSVersion 18.1
     *
     * Indicates that the time of the event has not been determined yet.
     * Leads to showing "TBD" in the UI when `date` is set.
     * Setting `ignoreTimeComponents` to true, has higher priority
     * over this property.
     *
     * This property has higher priority over `unannounced`.
     *
     * When both `date` and `semantics.eventStartDate` are unset,
     * `Date: TBD` will be shown in the UI.
     */
    undetermined: Joi.boolean(),
});
export const Location = Joi.object().keys({
    latitude: Joi.number().required(),
    longitude: Joi.number().required(),
});
export const PersonNameComponents = Joi.object().keys({
    givenName: Joi.string(),
    familyName: Joi.string(),
    middleName: Joi.string(),
    namePrefix: Joi.string(),
    nameSuffix: Joi.string(),
    nickname: Joi.string(),
    phoneticRepresentation: Joi.string(),
});
export const Seat = Joi.object().keys({
    seatSection: Joi.string(),
    seatRow: Joi.string(),
    seatNumber: Joi.string(),
    seatIdentifier: Joi.string(),
    seatType: Joi.string(),
    seatDescription: Joi.string(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     */
    seatAisle: Joi.string(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     */
    seatLevel: Joi.string(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     */
    seatSectionColor: Joi.string().regex(RGB_HEX_COLOR_REGEX),
});
export const WifiNetwork = Joi.object().keys({
    password: Joi.string().required(),
    ssid: Joi.string().required(),
});
