// passkit-generator/lib/esm/schemas/UpcomingPassInformation.js
import Joi from "joi";
import { PassFieldContent } from "./PassFieldContent.js";
import { Semantics } from "./Semantics.js";
import { URL_REGEX } from "./regexps.js";
const ImageURLEntry = Joi.object({
    SHA256: Joi.string().required(),
    URL: Joi.string().regex(URL_REGEX).required(),
    scale: Joi.number().default(1),
    size: Joi.number().max(2 * 1024 * 1024), // 2 megabytes max
});
const Image = Joi.object({
    URLs: Joi.array().items(ImageURLEntry),
    reuseExisting: Joi.boolean(),
});
const Images = Joi.object({
    headerImage: Image,
    venueMap: Image,
});
const URLs = Joi.object({
    accessibilityURL: Joi.string().regex(URL_REGEX),
    addOnURL: Joi.string().regex(URL_REGEX),
    bagPolicyURL: Joi.string().regex(URL_REGEX),
    /**
     * Joi's email schema validates email TLDs and only allows TLDs that are registered in the IANA Registry.
     * This also requires NodeJS runtime to work (even though can be enabled client side, but discouraged)
     * Reference - https://github.com/hapijs/joi/issues/2390
     */
    contactVenueEmail: Joi.string().email({ tlds: false }),
    contactVenuePhoneNumber: Joi.string(),
    contactVenueWebsite: Joi.string().regex(URL_REGEX),
    directionsInformationURL: Joi.string().regex(URL_REGEX),
    merchandiseURL: Joi.string().regex(URL_REGEX),
    orderFoodURL: Joi.string().regex(URL_REGEX),
    parkingInformationURL: Joi.string().regex(URL_REGEX),
    purchaseParkingURL: Joi.string().regex(URL_REGEX),
    sellURL: Joi.string().regex(URL_REGEX),
    transferURL: Joi.string().regex(URL_REGEX),
    transitInformationURL: Joi.string().regex(URL_REGEX),
});
const DateInformation = Joi.object({
    date: Joi.alternatives(Joi.string().isoDate(), Joi.date().iso()).required(),
    ignoreTimeComponents: Joi.boolean(),
    isAllDay: Joi.boolean(),
    isUnannounced: Joi.boolean(),
    isUndetermined: Joi.boolean(),
    timeZone: Joi.string(),
});
export const UpcomingPassInformationEntry = Joi.object({
    URLs: URLs,
    additionalInfoFields: Joi.array().items(PassFieldContent),
    auxiliaryStoreIdentifiers: Joi.array().items(Joi.number()),
    backFields: Joi.array().items(PassFieldContent),
    dateInformation: DateInformation,
    identifier: Joi.string().required(),
    images: Images,
    isActive: Joi.boolean(),
    name: Joi.string().required(),
    semantics: Semantics.concat(Joi.object({
        venuePlaceID: Joi.string(),
    })),
    type: Joi.string().valid("event").required(),
});
