// passkit-generator/lib/cjs/schemas/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Template = exports.PassProps = exports.OverridablePassProps = exports.PassType = exports.PassKindsProps = exports.PassPropsFromMethods = exports.RelevantDate = exports.PreferredStyleSchemes = void 0;
exports.assertValidity = assertValidity;
exports.validate = validate;
exports.filterValid = filterValid;
const tslib_1 = require("tslib");
tslib_1.__exportStar(require("./Barcode.js"), exports);
tslib_1.__exportStar(require("./Beacon.js"), exports);
tslib_1.__exportStar(require("./Location.js"), exports);
tslib_1.__exportStar(require("./PassFieldContent.js"), exports);
tslib_1.__exportStar(require("./NFC.js"), exports);
tslib_1.__exportStar(require("./Semantics.js"), exports);
tslib_1.__exportStar(require("./PassFields.js"), exports);
tslib_1.__exportStar(require("./Personalize.js"), exports);
tslib_1.__exportStar(require("./Certificates.js"), exports);
tslib_1.__exportStar(require("./UpcomingPassInformation.js"), exports);
const joi_1 = tslib_1.__importDefault(require("joi"));
const Barcode_js_1 = require("./Barcode.js");
const Location_js_1 = require("./Location.js");
const Beacon_js_1 = require("./Beacon.js");
const NFC_js_1 = require("./NFC.js");
const PassFields_js_1 = require("./PassFields.js");
const Semantics_js_1 = require("./Semantics.js");
const UpcomingPassInformation_js_1 = require("./UpcomingPassInformation.js");
const Messages = tslib_1.__importStar(require("../messages.js"));
const regexps_js_1 = require("./regexps.js");
exports.PreferredStyleSchemes = joi_1.default.array().items("posterEventTicket", "eventTicket", 
// or, since iOS 26
"boardingPass", "semanticBoardingPass");
exports.RelevantDate = joi_1.default.alternatives(joi_1.default.object().keys({
    startDate: joi_1.default.alternatives(joi_1.default.string().isoDate(), joi_1.default.date().iso()).required(),
    endDate: joi_1.default.alternatives(joi_1.default.string().isoDate(), joi_1.default.date().iso()).required(),
}), joi_1.default.object().keys({
    /**
     * Since iOS 26
     */
    date: joi_1.default.alternatives(joi_1.default.string().isoDate(), joi_1.default.date().iso()),
    /**
     * Since iOS 18, then was renamed in
     * 'date' in iOS 26 (what a breaking change)
     */
    relevantDate: joi_1.default.alternatives(joi_1.default.string().isoDate(), joi_1.default.date().iso()).required(),
}));
exports.PassPropsFromMethods = joi_1.default.object({
    nfc: NFC_js_1.NFC,
    beacons: joi_1.default.array().items(Beacon_js_1.Beacon),
    barcodes: joi_1.default.array().items(Barcode_js_1.Barcode),
    relevantDate: joi_1.default.string().isoDate(),
    relevantDates: joi_1.default.array().items(exports.RelevantDate),
    expirationDate: joi_1.default.string().isoDate(),
    locations: joi_1.default.array().items(Location_js_1.Location),
    preferredStyleSchemes: exports.PreferredStyleSchemes,
    upcomingPassInformation: joi_1.default.array().items(UpcomingPassInformation_js_1.UpcomingPassInformationEntry),
});
exports.PassKindsProps = joi_1.default.object({
    coupon: PassFields_js_1.PassFields.disallow("transitType"),
    generic: PassFields_js_1.PassFields.disallow("transitType"),
    storeCard: PassFields_js_1.PassFields.disallow("transitType"),
    eventTicket: PassFields_js_1.PassFields.disallow("transitType"),
    boardingPass: PassFields_js_1.PassFields,
});
exports.PassType = joi_1.default.string().regex(/(boardingPass|coupon|eventTicket|storeCard|generic)/);
exports.OverridablePassProps = joi_1.default.object({
    formatVersion: joi_1.default.number().default(1),
    semantics: Semantics_js_1.Semantics,
    voided: joi_1.default.boolean(),
    logoText: joi_1.default.string(),
    description: joi_1.default.string(),
    serialNumber: joi_1.default.string(),
    appLaunchURL: joi_1.default.string().uri(),
    teamIdentifier: joi_1.default.string(),
    organizationName: joi_1.default.string(),
    passTypeIdentifier: joi_1.default.string(),
    sharingProhibited: joi_1.default.boolean(),
    groupingIdentifier: joi_1.default.string(),
    suppressStripShine: joi_1.default.boolean(),
    maxDistance: joi_1.default.number().positive(),
    authenticationToken: joi_1.default.string().min(16),
    labelColor: joi_1.default.string().regex(regexps_js_1.RGB_HEX_COLOR_REGEX),
    stripColor: joi_1.default.string().regex(regexps_js_1.RGB_HEX_COLOR_REGEX),
    backgroundColor: joi_1.default.string().regex(regexps_js_1.RGB_HEX_COLOR_REGEX),
    foregroundColor: joi_1.default.string().regex(regexps_js_1.RGB_HEX_COLOR_REGEX),
    associatedStoreIdentifiers: joi_1.default.array().items(joi_1.default.number()),
    userInfo: joi_1.default.alternatives(joi_1.default.object().unknown(), joi_1.default.array()),
    webServiceURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    bagPolicyURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle posterEventTicket, semanticBoardingPasses
     * @passDomain Event Guide, Semantic Boarding Passes
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    orderFoodURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    parkingInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    directionsInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * URL to a resource to buy or access
     * the parking spot.
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    purchaseParkingURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * URL to a resource to buy the
     * merchandise.
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    merchandiseURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * URL to a resource about public or
     * private transportation to reach the
     * venue.
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    transitInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * URL to a resource about accessibility
     * in the events venue.
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    accessibilityURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * An URL to link experiences to the
     * pass (upgrades and more).
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    addOnURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    contactVenueEmail: joi_1.default.string(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    contactVenuePhoneNumber: joi_1.default.string(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain Event Guide
     *
     * @description
     *
     * To show buttons in the event guide,
     * at least two among those marked with
     * "@passDomain Event Guide" must be used.
     */
    contactVenueWebsite: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * Will add a button among options near "share"
     */
    transferURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * Will add a button among options near "share"
     */
    sellURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * Will remove an automatic shadow in the new
     * event ticket layouts.
     */
    suppressHeaderDarkening: joi_1.default.boolean(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * By default, the chin is colored with a
     * blur. Through this option, it is possible
     * to specify a different and specific color
     * for it.
     */
    footerBackgroundColor: joi_1.default.string().regex(regexps_js_1.RGB_HEX_COLOR_REGEX),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * Enables the automatic calculation of the
     * `foregroundColor` and `labelColor` based
     * on the background image in the new event
     * ticket passes.
     *
     * If enabled, `foregroundColor` and `labelColor`
     * are ignored.
     */
    useAutomaticColors: joi_1.default.boolean(),
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     *
     * @description
     *
     * Applications AppStore Identifiers
     * related to the event ticket.
     *
     * It is not mandatory for the app to
     * be related to the pass issuer.
     *
     * Such applications won't be able to read
     * the passes users has (probably differently
     * by `associatedStoreIdentifiers`).
     */
    auxiliaryStoreIdentifiers: joi_1.default.array().items(joi_1.default.number()),
    /**
     * @iOSVersion 18.1
     *
     * The text to display next to the logo on posterEventTicket passes.
     */
    eventLogoText: joi_1.default.string(),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for changing the seat for the ticket.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    changeSeatURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for in-flight entertainment.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    entertainmentURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for adding checked bags for the ticket.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    purchaseAdditionalBaggageURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL that links to information to purchase lounge access.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    purchaseLoungeAccessURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for purchasing in-flight wifi.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    purchaseWifiURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for upgrading the flight.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    upgradeURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for management.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    managementURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL for registering a service animal.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    registerServiceAnimalURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL to report a lost bag.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    reportLostBagURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * A URL to request a wheel chair.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    requestWheelchairURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * The email for the transit provider.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    transitProviderEmail: joi_1.default.string(),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * The phone number for the transit provider.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    transitProviderPhoneNumber: joi_1.default.string(),
    /**
     * @iOSVersion 26
     *
     * @description
     *
     * The URL for the transit provider.
     * Available only with Enhanced (or semantic) Boarding Passes
     */
    transitProviderWebsiteURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
}).with("webServiceURL", "authenticationToken");
exports.PassProps = joi_1.default.object()
    .concat(exports.OverridablePassProps)
    .concat(exports.PassKindsProps)
    .concat(exports.PassPropsFromMethods);
exports.Template = joi_1.default.object({
    model: joi_1.default.string().required(),
    certificates: joi_1.default.object().required(),
});
// --------- UTILITIES ---------- //
/**
 * Performs validation of a schema on an object.
 * If it fails, will throw an error.
 *
 * @param schema
 * @param data
 */
function assertValidity(schema, data, customErrorMessage) {
    const validation = schema.validate(data);
    if (validation.error) {
        if (customErrorMessage) {
            console.warn(validation.error);
            throw new TypeError(`${validation.error.name} happened. ${Messages.format(customErrorMessage, validation.error.message)}`);
        }
        throw new TypeError(validation.error.message);
    }
}
/**
 * Performs validation and throws the error if there's one.
 * Otherwise returns a (possibly patched) version of the specified
 * options (it depends on the schema)
 *
 * @param schema
 * @param options
 * @returns
 */
function validate(schema, options) {
    const validationResult = schema.validate(options, {
        stripUnknown: true,
        abortEarly: true,
    });
    if (validationResult.error) {
        throw validationResult.error;
    }
    return validationResult.value;
}
function filterValid(schema, source) {
    if (!source) {
        return [];
    }
    return source.reduce((acc, current) => {
        try {
            return [...acc, validate(schema, current)];
        }
        catch (err) {
            console.warn(Messages.format(Messages.FILTER_VALID.INVALID, err));
            return [...acc];
        }
    }, []);
}
