// passkit-generator/lib/cjs/schemas/PassFields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PassFields = exports.TransitType = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const PassFieldContent_js_1 = require("./PassFieldContent.js");
exports.TransitType = joi_1.default.string().regex(/(PKTransitTypeAir|PKTransitTypeBoat|PKTransitTypeBus|PKTransitTypeGeneric|PKTransitTypeTrain)/);
exports.PassFields = joi_1.default.object().keys({
    auxiliaryFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContentWithRow),
    backFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    headerFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    primaryFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    secondaryFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    transitType: exports.TransitType,
    /**
     * @iOSVersion 18
     * @passStyle eventTicket (new layout)
     * @passDomain dashboard
     *
     * @see \<undiclosed>
     */
    additionalInfoFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
});
