// passkit-generator/lib/cjs/schemas/UpcomingPassInformation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpcomingPassInformationEntry = void 0;
const tslib_1 = require("tslib");
const joi_1 = tslib_1.__importDefault(require("joi"));
const PassFieldContent_js_1 = require("./PassFieldContent.js");
const Semantics_js_1 = require("./Semantics.js");
const regexps_js_1 = require("./regexps.js");
const ImageURLEntry = joi_1.default.object({
    SHA256: joi_1.default.string().required(),
    URL: joi_1.default.string().regex(regexps_js_1.URL_REGEX).required(),
    scale: joi_1.default.number().default(1),
    size: joi_1.default.number().max(2 * 1024 * 1024), // 2 megabytes max
});
const Image = joi_1.default.object({
    URLs: joi_1.default.array().items(ImageURLEntry),
    reuseExisting: joi_1.default.boolean(),
});
const Images = joi_1.default.object({
    headerImage: Image,
    venueMap: Image,
});
const URLs = joi_1.default.object({
    accessibilityURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    addOnURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    bagPolicyURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    /**
     * Joi's email schema validates email TLDs and only allows TLDs that are registered in the IANA Registry.
     * This also requires NodeJS runtime to work (even though can be enabled client side, but discouraged)
     * Reference - https://github.com/hapijs/joi/issues/2390
     */
    contactVenueEmail: joi_1.default.string().email({ tlds: false }),
    contactVenuePhoneNumber: joi_1.default.string(),
    contactVenueWebsite: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    directionsInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    merchandiseURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    orderFoodURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    parkingInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    purchaseParkingURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    sellURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    transferURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
    transitInformationURL: joi_1.default.string().regex(regexps_js_1.URL_REGEX),
});
const DateInformation = joi_1.default.object({
    date: joi_1.default.alternatives(joi_1.default.string().isoDate(), joi_1.default.date().iso()).required(),
    ignoreTimeComponents: joi_1.default.boolean(),
    isAllDay: joi_1.default.boolean(),
    isUnannounced: joi_1.default.boolean(),
    isUndetermined: joi_1.default.boolean(),
    timeZone: joi_1.default.string(),
});
exports.UpcomingPassInformationEntry = joi_1.default.object({
    URLs: URLs,
    additionalInfoFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    auxiliaryStoreIdentifiers: joi_1.default.array().items(joi_1.default.number()),
    backFields: joi_1.default.array().items(PassFieldContent_js_1.PassFieldContent),
    dateInformation: DateInformation,
    identifier: joi_1.default.string().required(),
    images: Images,
    isActive: joi_1.default.boolean(),
    name: joi_1.default.string().required(),
    semantics: Semantics_js_1.Semantics.concat(joi_1.default.object({
        venuePlaceID: joi_1.default.string(),
    })),
    type: joi_1.default.string().valid("event").required(),
});
